web/src/components/storage/iscsi/TargetsSection.test.tsx

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright (c) [2023] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import React from "react";
+import { screen, waitFor, within } from "@testing-library/react";
+import { plainRender } from "~/test-utils";
+import TargetsSection from "./TargetsSection";
+import { deleteNode, discover, login, logout } from "~/api/storage/iscsi";
+
+let mockNodes = [];
+const mockDiscover = jest.fn();
+const node1 = {
+  id: 1,
+  address: "192.168.122.10",
+  target: "iqn.2024-09.com.example",
+  port: "e5ce1696dbf39d938e88",
+  connected: false,
+  ibft: false,
+  interface: "default",
+  startup: "",
+};
+
+const node2 = {
+  id: 2,
+  address: "192.168.122.10",
+  target: "iqn.2024-09.com.example",
+  port: "a9fd1c74adea4cbd80502",
+  connected: true,
+  ibft: false,
+  interface: "default",
+  startup: "onboot",
+};
+
+jest.mock("~/queries/storage/iscsi", () => ({
+  ...jest.requireActual("~/queries/storage/iscsi"),
+  useNodes: () => mockNodes,
+}));
+
+jest.mock("~/api/storage/iscsi", () => ({
+  ...jest.requireActual("~/queries/storage/iscsi"),
```

Comment:
is there reason to load rest of queries? Cannot it do some backend communication? I am asking this in all those mock that include also requireActual

Reply:
For no particular reason. I like to keep the original mode and mock only those functions that the test will use.

Reply:
The part that worry me is that it can cause accidental communication with backend...if it is not there, it will immediatelly fail in test, not?

Reply:
Yes, the test will fail in either case.